frontend/src/pages/Chats.tsx

```diff
@@ -24,23 +25,28 @@ import './Chats.css';
 import { getChatClient, getChannel } from '../functions/chats';
 
 export const Chats = () => {
+  const { state } = useLocation();
+  const { user } = state;
+
   const [client, setClient] = useState<StreamChat<DefaultGenerics>>();
   const [channel, setChannel] = useState<StreamChannel<DefaultGenerics>>();
 
-  const user = {
-    id: 'john',
-    name: 'John',
-    image: 'https://getstream.imgix.net/images/random_svg/FS.png',
-  };
-
   useEffect(() => {
     async function init() {
       const chatClient = getChatClient();
 
-      // Dev Token for now, would need to use appSecret to create real userToken
-      await chatClient.connectUser(user, chatClient.devToken(user.id));
+      await chatClient.connectUser(
+        {
+          id: user.primaryKey.slice(5), // removes USER# prefix
+          name: user.name,
+          // image: '',
+        },
+        user.chatToken
+        // chatClient.devToken(user.primaryKey)
+      );
 
-      const channel = getChannel(chatClient, user.id);
+      // Note that creating a channel using userIDs requires at least two
+      const channel = getChannel(chatClient, [user.primaryKey.slice(5), 'john'], user.name);
```

Comment:
This is an interesting one. It seems that by default, users are put into a group with a fake user called 'john'. Rather than doing this, in `init`, is it possible to get all channels for a user. If channels.length === 0, then we can display a message that tells them to look out on Tuesday for their first match. 

If channels.length > 0, then we can setChannel to the last one